.circleci/config.yml

```diff
@@ -1,67 +1,20 @@
 version: 2.1
 
-orbs:
-  codecov: codecov/codecov@3.2.2
-
-jobs:
-  tests:
-    docker:
-      - image: 'cimg/python:3.10'
-        environment:
-          TOX_POSARGS: ''
-      - image: 'docker.elastic.co/elasticsearch/elasticsearch:7.14.0'
-        name: search
-        environment:
-          discovery.type: single-node
-          ES_JAVA_OPTS: -Xms750m -Xmx750m
-    steps:
-      - checkout
-      - run: git submodule sync
-      - run: git submodule update --init
-      - run: pip install --user tox
-      - run: tox -e py310
-      - codecov/upload
+setup: true
 
-  tests-embedapi:
-    docker:
-      - image: 'cimg/python:3.10'
-    steps:
-      - checkout
-      - run: git submodule sync
-      - run: git submodule update --init
-      - run: pip install --user tox
-      - run: tox -c tox.embedapi.ini
-
-  checks:
-    docker:
-      - image: 'cimg/python:3.10'
-        environment:
-          NODE_VERSION: 10.17.0
-    steps:
-      - checkout
-      - run: git submodule sync
-      - run: git submodule update --init
-      - run: pip install --user tox
-      - run: scripts/circle/install_node.sh
-      - run:
-          name: Add node to the path
-          command: |
-            echo 'export PATH=~/.nvm/versions/node/v${NODE_VERSION}/bin:$PATH' >> $BASH_ENV
-            source $BASH_ENV
-      - run: tox -e migrations
-      - run: tox -e pre-commit
-      - run: tox -e lint
-      - run: tox -e docs
-      - run: tox -e docs-dev
-      - run: tox -e eslint
+orbs:
+  path-filtering: circleci/path-filtering@0.1.3
 
 workflows:
-  version: 2
-  test:
+  generate-config:
     jobs:
-      - checks
-      - tests
-      - tests-embedapi:
-          requires:
-            - checks
-            - tests
+      - path-filtering/filter:
+          base-revision: main
+          config-path: .circleci/config-conditional-builds.yml
+          mapping: |
+            requirements/.* build-code true
+            readthedocs/.* build-code true
+            pytest.* build-code true
+            tox.* build-code true
+            package.* build-code true
+            docs/.* build-docs true
```

Comment:
This part can be removed if https://github.com/readthedocs/readthedocs.org/pull/9788 is merged.